packages/x-date-pickers/src/internals/hooks/useField/useFieldState.ts

```diff
@@ -242,6 +242,7 @@ export const useFieldState = <
     return setState((prevState) => ({
       ...prevState,
       sections: newSections,
+      tempValueStrAndroid: null,
```

Comment:
Makes sure that `cleanActiveSection` resets the `tempValueStrAndroid`
We could probably replace all the `setState` in this file with a wrapper which removes `tempValueStrAndroid` to avoid having to define this `null` value each time.